File: backend/src/main/java/com/pwr/psi/backend/model/dto/ThesisDto.java
package com.pwr.psi.backend.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class ThesisDto {
    private String theme;
    private String supervisorUsername;
    private String supervisorNames;
    private String type;
    private String year;
    private String fieldName;
    private String language;
    private String status;
    private List<UserDto> authors;
    private int thesisId;
    private String reporter;
}

File: frontend/src/app/model/dto/thesisDto.ts
import { UserDto } from "./user";

export class ThesisDto {
  theme: string;
  supervisorUsername: string;
  supervisorNames: string;
  type: string;
  year: string;
  fieldName: string;
  language: string;
  status: string;
  authors:UserDto[];
  thesisId:number;;
  reporter:string;
}
